src/coreComponents/constitutive/relativePermeability/RelativePermeabilityInterpolators.hpp

```diff
@@ -0,0 +1,114 @@
+   * @param[in] shiftedWaterVolFrac
+   * @param[in] gasVolFrac
+   * @param[out] threePhaseRelPerm
+   * @param[out] dThreePhaseRelPerm_dVolFrac
+   * @param[in] relPerm_wo
+   * @param[in] dRelPerm_wo_dOilVolFrac
+   * @param[in] relPerm_go
+   * @param[in] dRelPerm_go_dOilVolFrac
+   *
+   * This function interpolates the two-phase relperms to compute the three-phase relperm
+   * The interpolation is based on the modified Baker method, also used as default in Eclipse
+   * Reference: Eclipse technical description and PetroWiki
+   */
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  static void
+  compute( real64 const & shiftedWaterVolFrac,
+           real64 const & gasVolFrac,
+           arraySlice1d< integer const > const & phaseOrder,
+           real64 const & relPerm_wo,
+           real64 const & dRelPerm_wo_dOilVolFrac,
+           real64 const & relPerm_go,
+           real64 const & dRelPerm_go_dOilVolFrac,
+           real64 & threePhaseRelPerm,
+           arraySlice1d< real64 > const & dThreePhaseRelPerm_dVolFrac )
+  {
+    integer const ip_water = phaseOrder[RelativePermeabilityBase::PhaseType::WATER];
+    integer const ip_oil   = phaseOrder[RelativePermeabilityBase::PhaseType::OIL];
+    integer const ip_gas   = phaseOrder[RelativePermeabilityBase::PhaseType::GAS];
+
+    // if water phase is immobile, then use the two-phase gas-oil data only
+    if( shiftedWaterVolFrac < LvArray::NumericLimits< real64 >::epsilon )
+    {
+      threePhaseRelPerm = relPerm_go;
+      dThreePhaseRelPerm_dVolFrac[ip_oil] = dRelPerm_go_dOilVolFrac;
+    }
+    // if gas phase is immobile, then use the two-phase water-oil data only
+    else if( gasVolFrac < LvArray::NumericLimits< real64 >::epsilon )
+    {
+      threePhaseRelPerm = relPerm_wo;
+      dThreePhaseRelPerm_dVolFrac[ip_oil] = dRelPerm_wo_dOilVolFrac;
+    }
+    // if both the water phase and the gas phase are mobile,
+    // then use a saturation-weighted interpolation of the two-phase oil rel perms
+    else
+    {
+      real64 const sumRelPerm = (shiftedWaterVolFrac * relPerm_wo
+                                 + gasVolFrac   * relPerm_go);
+      real64 const dSumRelPerm_dWaterVolFrac = relPerm_wo;
+      real64 const dSumRelPerm_dOilVolFrac   = shiftedWaterVolFrac * dRelPerm_wo_dOilVolFrac
+                                               + gasVolFrac   * dRelPerm_go_dOilVolFrac;
+      real64 const dSumRelPerm_dGasVolFrac   = relPerm_go;
+
+
+      real64 const sumVolFrac    = shiftedWaterVolFrac + gasVolFrac;
+      real64 const sumVolFracInv = 1 / sumVolFrac; // div by 0 handled by the if statement above
+      real64 const dSumVolFracInv_dWaterVolFrac = -sumVolFracInv * sumVolFracInv;
+      real64 const dSumVolFracInv_dGasVolFrac   = dSumVolFracInv_dWaterVolFrac;
+
+      threePhaseRelPerm = sumRelPerm * sumVolFracInv; // three-phase oil rel perm
+      dThreePhaseRelPerm_dVolFrac[ip_water] = dSumRelPerm_dWaterVolFrac * sumVolFracInv // derivative w.r.t. Sw
```

Comment:
Here I moved the comments above the assignments as you suggest. 
But in general I really like to have the derivatives on two lines because visually it helps me debug/check the indices (even though `uncrustify` does not align the expressions exactly as I would like to so they are a little bit shifted).